### PR TITLE
use transfer event instead coin_received & coin_spent

### PR DESCRIPTION
**Pull request by nimate**

### CODE REVIEW

**Comment by tasinco** [comment]
remove `groupRSEvents` -- it's not used?

**Comment by tasinco** [comment]
is this needed? https://github.com/figment-networks/ni-cosmoslib/blob/d5958354c7f46c8936bae6a2da78587a3d01a6a6/api/mapper/reward/reward.go#L339

File: api/mapper/reward/reward.go [hunk 1]
```diff
@@ -333,16 +329,16 @@ func (m *Mapper) MsgBeginRedelegate(msg []byte, lg types.ABCIMessageLog) (rev *r
...
-							This code is temporary and may not work in some cases.
-						*/
-						if i%2 == 0 { // TODO in orde
-							reward.Validator = wvc.ValidatorSrcAddress
-						} else {
-							reward.Validator = wvc.ValidatorDstAddress
-						}
-						if wvc.DelegatorAddress != p["receiver"] {
-							rev.RewardRecipients = append(rev.RewardRecipients, p["receiver"])
+						// /*	To be sure which transaction belongs to which validator, we need to check the delegation for the previous block.
```
Comment (tasinco) [inline api/mapper/reward/reward.go:332]: just remove it?

**Review by tasinco — APPROVED** [review]

File: api/mapper/reward/reward.go (hunk 1 above)
Comment (nimate) [reply api/mapper/reward/reward.go:332]: sure, I thought I already deleted that

**Review by greywolve — APPROVED** [review]

**Pull request merged**